spec/apps/mpns_app_spec.rb

```diff
@@ -0,0 +1,75 @@
+module RailsPushNotifications
+  describe MPNSApp, type: :model do
+
+    it 'creates an instance given valid attributes' do
+      MPNSApp.create! attributes_for :mpns_app
+    end
+
+    describe 'validations' do
+
+      let(:app) { build :mpns_app }
+
+      it 'requires the certificate' do
+        app.cert = nil
+        expect(app).to_not be_valid
+      end
+    end
+
+    describe 'notifications relationship' do
+
+      let(:app) { create :mpns_app }
+
+      it 'can create new notifications' do
+        expect do
+          app.notifications.create attributes_for :mpns_notification
+        end.to change { app.reload.notifications.count }.from(0).to 1
+      end
+    end
+
+    describe '#push_notifications' do
+
+      let(:app) { create :mpns_app }
+      let(:notifications) {
+        (1..10).map { create :mpns_notification, app: app }
+      }
+      let(:single_notification) { create :mpns_notification, app: app }
+
+      let(:headers) {
+          {
+            'x-notificationstatus' => 'Received',
+            'x-deviceconnectionstatus' => 'Connected',
+            'x-subscriptionstatus' => 'Active'
+          }
+        }
+      let(:response) { [ { device_url: 'http://s.notify.live.net/1', headers: headers, code: 200 } ]}
```

Comment:
Line is too long. [101/80]<br>Space missing inside }.<br>Space inside square brackets detected.